build-tools/update

```diff
@@ -36,9 +43,22 @@ try {
 	version += ` (${head.slice(0, 8)}${head !== origin ? `/${origin.slice(0, 8)}` : ''})`;
 } catch (e) {}
 
-let configBuf = fs.readFileSync('config/config.js', {encoding: 'utf8'});
-configBuf = configBuf.replace(/\/\* version \*\/[^;\n]*;/, `/* version */ Config.version = ${JSON.stringify(version)};`);
-
+const routes = JSON.parse(fs.readFileSync('config/routes.json'));
+const autoconfigRegex = new RegExp(`${escapeRegex(AUTOCONFIG_START)}[^]+${escapeRegex(AUTOCONFIG_END)}`, 'g');
+const autoconfig = `${AUTOCONFIG_START}
+Config.version = ${JSON.stringify(version)};
+
+Config.routes = {
+	root: '${routes.root}',
+	client: '${routes.client}',
+	dex: '${routes.dex}',
+	replays: '${routes.replays}',
+	users: '${routes.users}',
+};
+${AUTOCONFIG_END}`;
+
+// remove old automatically generated configuration and add the new one
+const configBuf = fs.readFileSync('config/config.js', {encoding: 'utf8'}).replace(autoconfigRegex, '') + autoconfig;
```

Comment:
```suggestion
const configBuf = fs.readFileSync('config/config.js', {encoding: 'utf8'}).replace(autoconfigRegex, autoconfig);
```

The way the autoconfig is currently done, it's non-obvious that you can't put stuff after it. This will let you put stuff after it.

Reply:
Oh, I see, appending is a special-case in case the regex didn't find anything.

I think it's worth incurring a double-search for this (annoying that there's no way for `string.replace` to detect if the replace succeeded or failed).

Reply:
```suggestion
let configBuf = fs.readFileSync('config/config.js', {encoding: 'utf8'});
if (autoconfigRegex.test(configBuf)) {
  configBuf = configBuf.replace(autoconfigRegex, autoconfig);
} else {
  configBuf += autoconfig;
}
```